### PR TITLE
doc: add readme for on-chain sig support

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

add readme. closes https://github.com/MystenLabs/sui/issues/11460

## Test Plan 

n/a

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/learn/cryptography/sui-move-signatures.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+title: Sui On-Chain Signatures Verification in Move
+---
+
+Move contracts in Sui supports verifications for several signature schemes on chain. Note that not all signatures supported in on-chain verification are supported as user signature verification. See [Sui Signatures](sui-signatures.md#user-signature) for valid signatures to be used for transaction execution.
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-move-signatures.md:5]: 1. support (no s)
2. for valid signatures -> for valid signature schemes for transaction authorization.

File: doc/src/learn/cryptography/sui-move-signatures.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let recovered = ecdsa_r1::secp256r1_ecrecover(&sig, &msg);
+    assert!(pk == recovered, 0);
+```
+
+ 1. BLS G1 signature (minSig setting)
```
Comment (kchalkias) [inline doc/src/learn/cryptography/sui-move-signatures.md:146]: are we going to add example for minPK setting as well?

File: doc/src/learn/cryptography/sui-move-signatures.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme ed25519 --msg $MSG --secret-key  $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's verify method. All inputs are represented in bytes in hex format:
```
Comment (randall-Mysten) [inline doc/src/learn/cryptography/sui-move-signatures.md:44]: is this part of step 3 or a new step? It is hard to tell with the current formatting. Is the 1. in the section above actually step 3, or a new step 1? It renders as 3

File: doc/src/learn/cryptography/sui-move-signatures.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let verify = ed25519::ed25519_verify(&sig, &pk, &msg);
+    assert!(verify == true, 0);
+```
+
+ 1. Secp256k1 non-recoverable signature (64 bytes)
```
Comment (randall-Mysten) [inline doc/src/learn/cryptography/sui-move-signatures.md:54]: typically we don't number single steps - is this a step in a large procedure or a separate step/task to complete? Maybe make it bold to indicate a new section, or a heading 3.

File: doc/src/learn/cryptography/sui-move-signatures.md (hunk 2 above)
Comment (joyqvq) [reply doc/src/learn/cryptography/sui-move-signatures.md:146]: yeah need to merge fastcrypto https://github.com/MystenLabs/fastcrypto/pull/579 first

File: doc/src/learn/cryptography/sui-move-signatures.md (hunk 3 above)
Comment (joyqvq) [reply doc/src/learn/cryptography/sui-move-signatures.md]: i edited it to make it clearer. 

there are two subpoints under `1. Ed25519 signature (64 bytes)`: `* Generate a key and sign a message. ` and `* Call the verify method in Move.`

same for all signature schemes, feel free to edit if its still not clear.

File: doc/src/learn/cryptography/sui-move-signatures.md (hunk 4 above)
Comment (joyqvq) [reply doc/src/learn/cryptography/sui-move-signatures.md]: this is not a step, they represents all signature scheme covered. i changed them to #### as a subsection. the two steps are `1. Generate a key and sign a message.`  and `2. Call the verify method in Move`. please revise if needed.

**Review by kchalkias — APPROVED** [review]

**Pull request merged**